catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -67,6 +72,16 @@
 
   private static final Security SECURITY = Security.getInstance();
 
+  private long transformerCheckPeriodMillis = TimeUnit.SECONDS.toMillis(30);
+
+  private long transformerWaitTimeoutMillis = TimeUnit.MINUTES.toMillis(3);
+
+  private static InputTransformerIds inputTransformerIds;
+
+  private final AtomicBoolean configDeleted = new AtomicBoolean(false);
+
+  private final Object lock = new Object();
```

Comment:
❓  Curious why a `ReentrantLock` couldn't be used here

Reply:
I wasn't aware of it and didn't consider it.

Reply:
I am not sure it would make the code any easier in this case.